pandas/tests/reshape/merge/test_merge_anti.py

```diff
@@ -0,0 +1,502 @@
+                "anti_left",
+                ["B", "C"],
+            ),
+            (
+                DataFrame(
+                    {"A": [np.nan, np.nan], "B": [5, 9], "C": [4, 7], "D": ["a", "d"]}
+                ),
+                "anti_right",
+                ["B", "C"],
+            ),
+            (
+                DataFrame(
+                    {
+                        "A": [1, 3, np.nan, np.nan],
+                        "B": [4, 6, 5, 9],
+                        "C": [5, 7, 4, 7],
+                        "D": [np.nan, np.nan, "a", "d"],
+                    }
+                ),
+                "anti_full",
+                ["B", "C"],
+            ),
+        ],
+    )
+    def test_anti_multicol(self, expected, how, on):
+        # test with multicol with and w/o on param
+        right = DataFrame({"B": [5, 5, 9], "C": [4, 6, 7], "D": ["a", "b", "d"]})
+        left = DataFrame(
+            {"A": [1, 2, 3], "B": [4, 5, 6], "C": [5, 6, 7]}, index=["a", "b", "c"]
+        )
+        result = merge(left, right, how=how, on=on)
+        tm.assert_frame_equal(result, expected)
+
+    @pytest.mark.parametrize(
+        "expected, how, on, left_on, right_on",
+        [
+            (
+                DataFrame({"A": np.nan, "B": "c", "C": np.nan}, index=[0]),
+                "anti_right",
+                None,
+                None,
+                None,
+            ),
+            (
+                DataFrame({"A": np.nan, "B": 3, "C": np.nan}, index=[0]).astype(
+                    {"B": object}
+                ),
+                "anti_left",
+                None,
+                None,
+                None,
+            ),
+            (
+                DataFrame({"A": np.nan, "B": "c", "C": np.nan}, index=[0]),
+                "anti_right",
+                ["B"],
+                None,
+                None,
+            ),
+            (
+                DataFrame({"A": np.nan, "B": 3, "C": np.nan}, index=[0]).astype(
+                    {"B": object}
+                ),
+                "anti_left",
+                ["B"],
+                None,
+                None,
+            ),
+            (
+                DataFrame(
+                    {"A": [2.0], "B_x": [2], "C": [np.nan], "B_y": [np.nan]}
+                ).astype({"B_x": object, "B_y": object}),
+                "anti_left",
+                None,
+                ["A"],
+                ["C"],
+            ),
+            (
+                DataFrame(
+                    {
+                        "A": [np.nan, np.nan],
+                        "B_x": [np.nan, np.nan],
+                        "C": [1.0, 3.0],
+                        "B_y": ["a", 2],
+                    }
+                ).astype({"B_x": object}),
+                "anti_right",
+                None,
+                ["A"],
+                ["C"],
+            ),
+        ],
+    )
+    def test_anti_with_nan(self, expected, how, on, left_on, right_on):
```

Comment:
what is this test doing that the other tests above do not capture?

Reply:
this was for mixed dtypes.
(added couple of more comments for these)